rbc/tests/test_omnisci_shared_dictionary.py

```diff
@@ -0,0 +1,115 @@
+import pytest
+from rbc.tests import omnisci_fixture
+
+
+@pytest.fixture(scope="module")
+def omnisci():
+
+    for o in omnisci_fixture(globals(), minimal_version=(5, 7)):
+        define(o)
+        yield o
+
+
+def define(omnisci):
+    @omnisci("int32(Column<int32>, RowMultiplier, OutputColumn<int32>)")
+    def test_shared_dict(x, m, y):
+        sz = len(x)
+        for i in range(sz):
+            y[i] = 1000 * x.get_dict_id() + x[i]
+        return m * sz
+
+    @omnisci("int32(Column<int32>, RowMultiplier, OutputColumn<bool>)")
+    def test_shared_dict_is_dict_encoded(x, m, y):
+        sz = len(x)
+        for i in range(sz):
+            y[i] = x.is_dict_encoded()
+        return m * sz
+
+
+@pytest.fixture(scope="function")
+def create_columns(omnisci):
+    omnisci.require_version(
+        (5, 7), "Requires omniscidb-internal PR 5492", date=20210531
+    )
```

Comment:
Note for myself: wait until omnisci/omniscidb-internal#5492 gets merged and make sure this works with the omnisci-latest image we have on CI.